notes/src/main/kotlin/com/ritense/note/service/NoteService.kt

```diff
@@ -24,57 +24,56 @@ import com.ritense.note.domain.Note
 import com.ritense.note.event.NoteCreatedEvent
 import com.ritense.note.event.NoteDeletedEvent
 import com.ritense.note.event.NoteUpdatedEvent
-import com.ritense.note.exception.NoteAccessDeniedException
 import com.ritense.note.exception.NoteNotFoundException
 import com.ritense.note.repository.NoteRepository
-import com.ritense.note.repository.SpecificationHelper
 import com.ritense.valtimo.contract.authentication.UserManagementService
-import com.ritense.valtimo.contract.utils.SecurityUtils
+import java.util.UUID
 import mu.KotlinLogging
 import org.springframework.context.ApplicationEventPublisher
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Pageable
-import java.util.UUID
+import org.springframework.transaction.annotation.Transactional
 
+@Transactional
 class NoteService(
     private val noteRepository: NoteRepository,
     private val userManagementService: UserManagementService,
     private val applicationEventPublisher: ApplicationEventPublisher,
     private val authorizationService: AuthorizationService,
 ) {
 
+    @Transactional(readOnly = true)
     fun getNotes(
         documentId: UUID,
         pageable: Pageable = Pageable.unpaged(),
     ): Page<Note> {
         val spec = authorizationService.getAuthorizationSpecification(
             AuthorizationRequest(
                 Note::class.java,
-                NoteActionProvider.VIEW
+                NoteActionProvider.LIST_VIEW
                 // TODO: The action type is not actually checked when requesting the spec directly
             ),
             null
         )
-        return noteRepository.findAll(spec.and(SpecificationHelper.byDocumentId(documentId)), pageable)
+        return noteRepository.findAll(spec, pageable)
```

Comment:
byDocumentId has to be in, otherwise you'll get results from other documents as well.